### PR TITLE
Adds {__eq__, __repr__} to {SwapPermutationGate, LinearPermutationGate}

### PR DESCRIPTION
**Pull request by bryano**

### CODE REVIEW

File: cirq/contrib/acquaintance/gates.py [hunk 1]
```diff
@@ -311,5 +311,11 @@ def permutation(self, qubit_count: int) -> Dict[int, int]:
         return {i: j for i, j in
                 enumerate(reversed(range(sum(self.part_lens))))}
 
+    def __repr__(self):
+        return ('cirq.contrib.acquaintance.SwapNetworkGate('
+                '{!r}, {!r})'.format(self.part_lens, self.acquaintance_size))
 
+    def __eq__(self, other):
```
Comment (Strilanc) [inline cirq/contrib/acquaintance/gates.py]: You also need `__ne__` and `__hash__`. And the type check.

Alternatively, you can use the `cirq.value_equality` decorator and implement `_value_equality_values_(self)`.

File: cirq/contrib/acquaintance/gates_test.py [hunk 2]
```diff
@@ -221,11 +226,20 @@ def test_swap_network_permutation(part_lens, acquaintance_size):
             zip(range(n_qubits), reversed(range(n_qubits)))}
     assert gate.permutation(n_qubits) == expected_permutation
 
+
```
Comment (Strilanc) [inline cirq/contrib/acquaintance/gates_test.py]: Add a short equality test using `cirq.testing.EqualsTester`.

File: cirq/contrib/acquaintance/permutation_test.py [hunk 3]
```diff
@@ -78,24 +81,25 @@ def test_update_mapping():
...
-        assert gate.permutation(n_elements) == permutation
-        mapping = dict(zip(qubits, elements))
-        for swap in cirq.flatten_op_tree(cirq.decompose_once_with_qubits(
-                gate, qubits)):
-            assert isinstance(swap, cirq.GateOperation)
-            swap.gate.update_mapping(mapping, swap.qubits)
-        for i in range(n_elements):
-            p = permutation.get(elements[i], i)
-            assert mapping.get(qubits[p], elements[i]) == i
+@pytest.mark.parametrize('n_elements,n_permuted',
```
Comment (Strilanc) [inline cirq/contrib/acquaintance/permutation_test.py]: Add a short equality test using `cirq.testing.EqualsTester`.

Comment (Strilanc) [inline cirq/contrib/acquaintance/permutation_test.py]: Add a repr test

File: cirq/contrib/acquaintance/permutation.py [hunk 4]
```diff
@@ -125,6 +130,16 @@ def _decompose_(self, qubits: Sequence[ops.QubitId]) -> ops.OP_TREE:
                     yield swap_gate(*qubits[i:i+2])
                     mapping[i], mapping[i+1] = mapping[i+1], mapping[i]
 
+    def __repr__(self):
+        return ('cirq.contrib.acquaintance.LinearPermutationGate('
+                '{!r}, {!r})'.format(self._permutation, self.swap_gate))
+
+    def __eq__(self, other):
```
Comment (Strilanc) [inline cirq/contrib/acquaintance/permutation.py]: You also need `__ne__` and `__hash__`. Alternatively, you can use the `cirq.value_equality` decorator and implement `_value_equality_values_(self)`.

File: cirq/contrib/acquaintance/permutation_test.py [hunk 5]
```diff
@@ -78,24 +81,25 @@ def test_update_mapping():
...
+def test_linear_permutation_gate(n_elements, n_permuted):
+    qubits = [cirq.NamedQubit(s) for s in alphabet[:n_elements]]
+    elements = tuple(range(n_elements))
+    elements_to_permute = sample(elements, n_permuted)
+    permuted_elements = sample(elements_to_permute, n_permuted)
+    permutation = {e: p for e, p in
+                   zip(elements_to_permute, permuted_elements)}
+    cca.PermutationGate.validate_permutation(permutation, n_elements)
+    gate = cca.LinearPermutationGate(permutation)
+    ct.assert_equivalent_repr(gate)
```
Comment (bryano) [inline cirq/contrib/acquaintance/permutation_test.py]: There's a repr test here. Does it need to be separated out?

File: cirq/contrib/acquaintance/permutation_test.py (hunk 3 above)
Comment (bryano) [reply cirq/contrib/acquaintance/permutation_test.py]: Done.

File: cirq/contrib/acquaintance/gates.py (hunk 1 above)
Comment (bryano) [reply cirq/contrib/acquaintance/gates.py]: Done.

File: cirq/contrib/acquaintance/permutation.py (hunk 4 above)
Comment (bryano) [reply cirq/contrib/acquaintance/permutation.py]: Done.

File: cirq/contrib/acquaintance/gates_test.py (hunk 2 above)
Comment (bryano) [reply cirq/contrib/acquaintance/gates_test.py]: Done.

**Pull request merged**